docs/source/challenge-doc.rst

```diff
@@ -373,6 +372,23 @@ List of states variables:
         - Ankle angle
         - Ankle velocity
 
+Participants are expected to modify the state paramters of the OSL state machine to tackle this challenge. Given that there are multiple terrain types in the MyoChallenge,
+we provide a structure that allows 4 sets of state paramters to be uploaded to the state machine. Participants can then freely switch between paramter sets during each episode by calling the required function.
+
+More details of the functions are here below:
+
+- upload_osl_param(dict_of_dict) `upload_osl_param <https://github.com/MyoHub/myosuite/blob/main/myosuite/envs/myo/myochallenge/run_track_v0.py#L717>`__
+    - This function expects a dictionary of dictionary containing state parameter values
```

Comment:
Are those nested dictionaries?

Reply:
Yes. They are nested dictionaries. I was using a list as a top level structure initially, but thought it would be a better idea to use a consistent data type for key based indexing.